js/src/launchAsync.ts

```diff
@@ -94,21 +96,23 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
+        const exit = (): void => {
+            reset();
+
+            // Execute exit callback if we have one
+            if (options.onExit) {
+                options.onExit();
+            }
+        }
+
         // Reset variables
         reset();
 
         const messageHandler = (e: any): void => {
             if (!e || !e.data) { return; }
 
-            if (e.data === 'ImmersiveReader-Exit') {
-                reset();
-                window.removeEventListener('message', messageHandler);
-
-                // Execute exit callback if we have one
-                if (options.onExit) {
-                    options.onExit();
-                }
-            } else if (e.data === 'ImmersiveReader-ReadyForContent') {
+            if (e.data === 'ImmersiveReader-ReadyForContent') {
+                resetTimeout(); // reset the timeout once the reader page loads successfully. The Reader page will report further errors through PostMessage if there is an issue obtaining the ConentModel from the server
```

Comment:
This is the timeout behavior adjustment. if the reader page itself loads successfully, then we nullify the timeout. And from there, we let the GetContentModel call proceed without timeout. From there, any errors will be reported through PostMessage.

I did not remove the Options.Timeout, because I thought it still might be useful to let customers configure this, just in case.